### PR TITLE
fix(gemm): honor bound scratch in inverse-RoPE WO projection

### PR DESCRIPTION
**Pull request by tpurtell**
## Summary

The inverse-RoPE WO binding exposes caller-owned quantization, intermediate and output views, but execution can allocate replacement buffers. Thread the bound views through inverse-RoPE quantization, WO-A, intermediate quantization and WO-B, including the fused small-token path. Return the output backed by the caller's arena.

Resolve the inverse-RoPE route once when constructing a binding and retain it for execution. The functional and bound paths share the existing route rules, including Spark-only quantized WO-A output at 16 tokens and the DSV4 TP2 geometry at 9–15 tokens. That route writes directly into bound `tmp_q` without `alpha_one` or a standalone intermediate quantizer. The small-token WO-B/preclear decision is also retained in the binding.

## Validation

Upstream comparison: master `75ffee6375b0577ce2c8d6931ffacefda3ecbdd6`.

21 targeted cases pass on RTX PRO 6000 Blackwell and GB10 with CUTLASS DSL 4.6.2: 14 CPU binding/route-boundary cases and seven GPU cases. GPU coverage includes the compact 1/16-token geometry and DSV4 TP2 at 8/9/15/16/17 tokens. Tests verify the retained route, forbid route selection during bound execution, compare with the functional path, mutate inputs, poison/replay the output, and check arena pointers and stable allocated bytes. Deterministic routes compare exactly; the DSV4 8-token BF16 atomic split-K control uses relative L2 < 0.005 and cosine > 0.99998 because accumulation order can vary. The original compact arena cases fail on upstream.

```bash
python -m pytest -o addopts= -p no:cacheprovider -q tests/gemm/test_wo_projection_scratch_bindings.py
python -m pytest -o addopts= -p no:cacheprovider -q tests/gemm/test_gemm_wo_projection.py -k inv_rope_planned_wo
```

On GB10, the existing `test_wo_inv_rope_route_fused_small_m_matches_reference_shapes` fails an exact-equality assertion on both upstream and this branch (maximum absolute difference 0.00390625); the targeted arena tests pass. This existing failure is outside the targeted selection above. No performance claim.

## Summary

`wo_projection_inv_rope_mxfp8` uses caller-owned scratch, quantization, intermediate, and output views for bound execution. Route selection is fixed during binding and covers atomic output clearing and quantized WO-A intermediates. This preserves arena pointers through inverse-RoPE quantization, WO-A, intermediate quantization, WO-B, and fused small-token execution.

Unbound execution retains the functional path. Bound outputs remain backed by the caller’s arena, satisfying the plan/bind/run storage contract required for CUDA graph replay.

## Validation

- Six CPU binding tests pass.
- GPU tests for 1 and 16 tokens pass on RTX PRO 6000 Blackwell and GB10.
- Tests verify arena pointers, reference results, and CUDA graph replay.
- Six selected inverse-RoPE GPU tests pass on RTX.
- An existing GB10 fused small-token test fails exact equality on both branches, with a maximum absolute difference of `0.00390625`.

### CODE REVIEW

File: b12x/gemm/_shared/wo_mxfp8.py [hunk 1]
```diff
@@ -3030,10 +3040,62 @@ def wo_projection_inv_rope_mxfp8(
...
+            quantize_wo_b_input_mxfp8(tmp, out=tmp_q)
+            wo_b_dense_gemm_mxfp8(
+                tmp_q,
+                weights.wo_b,
+                out=output,
+                alpha=alpha_one,
+                expected_m=expected_m,
+                sfb_k_replicated=weights.sfb_k_replicated,
+                stream=stream,
+            )
```
[the remark this answers is not in the document]
Comment (tpurtell) [reply b12x/gemm/_shared/wo_mxfp8.py]: Fixed in 19f4793c. Functional and bound inverse-RoPE paths now share the existing Spark B16 / DSV4 TP2 B9–15 predicate. The selected route is retained in WOProjectionInvRopeBinding, including the WO-B/preclear decision, and execution consumes it without rerunning selection. The quantized-intermediate route writes WO-A directly to bound tmp_q and omits alpha_one and the standalone intermediate quantizer.

Validation: 21 targeted cases pass on RTX PRO 6000 Blackwell and GB10 (14 CPU binding/route-boundary cases plus seven GPU cases). GPU tests cover compact M1/M16 and DSV4 TP2 M8/M9/M15/M16/M17, forbid route selection during bound execution, mutate inputs, poison/replay output, and preserve output pointers and allocated bytes. Deterministic routes compare exactly; the BF16 atomic M8 control uses explicit numerical tolerances. Ruff and git diff --check pass. The PR description includes the updated selection and limitations.
